### ISSUE TITLE
Temporary artifact source name collision 

### ISSUE THREAD

**Issue by fabiancook** [serverless/serverless#12749]
### Issue description

On generation of the ajv validation schemas  I receive this error when I try to deploy multiple services at once in one composed app:

```
ENOENT: no such file or directory, rename '/tmp/node-process-7687/f50acbb09eab13ac5b4b1f32951491cee7b22117.js' -> '/home/runner/.serverless/artifacts/ajv-validate-13-08-2024/f50acbb09eab13ac5b4b1f32951491cee7b22117.js'
```

It appears this is an issue with the source directory, not the target, as I have ensured the target is available beforehand and can confirm locally it is retained. 

It could be that it is possible for the names of a temporary directory to collide when multiple services are composed together. An alternative is that the directory is being deleted before the rename happens. 

Here is where the temporary name is created:

https://github.com/medikoo/process-utils/blob/60bfca5e7f286203c19a7d0eceb4e9728e0d9736/tmpdir/index.js#L15

This is what a temp directory ends up named like:

```
/tmp/node-process-afee 
```

Utilised here: 

https://github.com/serverless/serverless/blob/fec8e287743fcb9348827785079716fc91e8e6cb/lib/classes/config-schema-handler/resolve-ajv-validate.js#L81-L83 

### Temp  work around 

A solution is to make all of the services depend prior services to create a complete dependency chain, ensuring each service deploys one by one.

We would see the above error very often on deployment but was intermittently not showing up. 

I've created this issue here rather than in process-utils as I believe we should use a completely different temp directory name to truely solve this. 2 bytes might not be unique enough when you have a few services 

### Context

Its potential I am fully off here on the actual cause, it might not be a collision but the directory being deleted. Either way this code path is a bit suspect. 

N/A

**Comment by Mmarzex** [comment]
@fabiancook We have a fix for this ready and will be publishing a new version later today.

The issue is that that `process-utils`  method creates a single directory per process and in Compose a race condition was being caused. I removed this dependency and used standard `fs` methods to solve this.

**Comment by Mmarzex** [comment]
@fabiancook This has been fixed and published in version `4.1.22`, let me know if you run into any issues still.
